compilation error in fixed_buffer_stream.zig

Hi,

I got the following error:
```
/home/vincent/dev/devtools/zig/lib/std/io/fixed_buffer_stream.zig:70:37: error: expected type '[]u8', found '[]const u8'
            mem.copy(u8, self.buffer[self.pos .. self.pos + n], bytes[0..n]);
                                    ^
/home/vincent/dev/devtools/zig/lib/std/io/fixed_buffer_stream.zig:70:21: note: referenced here
            mem.copy(u8, self.buffer[self.pos .. self.pos + n], bytes[0..n]);
```

Here's a repro https://gcc.godbolt.org/z/vcn2ap

--- a/lib/std/io/stream_source.zig
+++ b/lib/std/io/stream_source.zig
@@ -39,7 +39,7 @@ pub const StreamSource = union(enum) {
     pub fn write(self: *StreamSource, bytes: []const u8) WriteError!usize {
         switch (self.*) {
             .buffer => |*x| return x.write(bytes),
-            .const_buffer => |*x| return x.write(bytes),
+            .const_buffer => return error.AccessDenied,
             .file => |x| return x.write(bytes),
         }
     }

